File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        array<long long, 2000> mapping;

        for (const auto& x : nums1)
            mapping[x]++;

        vector<int> intersection;
        for (const auto& x : nums2) {
            if (mapping[x] > 0) {
                intersection.push_back(x);
                mapping[x]--;
            }
        }
        return intersection;
    }
};

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.rs
impl Solution {
    pub fn intersect(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut mapping = [0; 2000];

        for &x in &nums1 {
            mapping[x as usize] += 1;
        }

        let mut intersection = Vec::new();
        for &x in &nums2 {
            if mapping[x as usize] > 0 {
                intersection.push(x);
                mapping[x as usize] -= 1;
            }
        }

        intersection
    }
}